### DISCUSSION TITLE
Session status when using navigate:true with return redirect() method.

### DISCUSSION THREAD

**Discussion by CodeWithPranta** [livewire/livewire#6278] [1. Help]
How I can pass Session status when using navigate:true with return redirect() method? I want to keep feeling like single page application. Not using modal so trying something like: `public function store() {
        $formFields = $this->validate([
            'title' => 'required|min:3',
            'content' => 'required|min:10',
        ]);

        Post::create($formFields);

        return $this->redirect('/posts', navigate: true)->with('status', 'Post done!');
    }`  but getting error: Call to a member function with() on null

**Comment by skeemer** [comment]
Checkout #6176, `$this->redirect()` is a void function. You will need to flash and then navigate.

```
$this->session->flash('status', 'Post done!');
$this->redirect('/posts', navigate: true);
```

**Reply by zacktagnan** [reply]
Maybe, better...
`session()->flash('status', 'Post done!');`

...to avoid an error like this: "_Property [$session] not found on component: [name-component]_"
